### PR TITLE
Refactor yet more utilities

### PR DESCRIPTION
**Pull request by zpqrtbnk**
Must merge #316 first

### CODE REVIEW

File: src/Hazelcast.Net/NetStandard/StreamExtensions.cs [hunk 1]
```diff
@@ -47,41 +48,53 @@ public static async ValueTask<int> ReadAsync(this Stream stream, Memory<byte> me
             if (stream == null) throw new ArgumentNullException(nameof(stream));
             if (memory.Length == 0) throw new ArgumentException("Empty memory.", nameof(memory));
 
-            byte[] bytes = null;
+            // this is based upon what 2.1 does
+
+            if (MemoryMarshal.TryGetArray(memory, out ArraySegment<byte> array))
+            {
+                //return await stream.ReadAsync(array.Array, array.Offset, array.Count, cancellationToken).CAF();
```
Comment (asimarslan) [inline src/Hazelcast.Net/NetStandard/StreamExtensions.cs:55]: is this a comment to show below workaround replacement? if so, can you make it no look like comment out code.

File: src/Hazelcast.Net/NetStandard/StreamExtensions.cs [hunk 2]
```diff
@@ -47,41 +48,53 @@ public static async ValueTask<int> ReadAsync(this Stream stream, Memory<byte> me
...
+                return result;
+            }
+
+            var bytes = ArrayPool<byte>.Shared.Rent(memory.Length);
             try
             {
-                // TODO: optimize
-                // of course this is sub-optimal :(
-                bytes = ArrayPool<byte>.Shared.Rent(memory.Length);
+                //var result = await stream.ReadAsync(bytes, 0, memory.Length, cancellationToken).CAF();
```
Comment (asimarslan) [inline src/Hazelcast.Net/NetStandard/StreamExtensions.cs:74]: same here as above comment

**Review by asimarslan — APPROVED** [review]
minor: two line of commented out code can be removed

File: src/Hazelcast.Net/NetStandard/StreamExtensions.cs (hunk 1 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net/NetStandard/StreamExtensions.cs:55]: same - this is what is replaced by the code under it - ok will try to make it more explicit

File: src/Hazelcast.Net/NetStandard/StreamExtensions.cs (hunk 2 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net/NetStandard/StreamExtensions.cs:74]: same, same

**Pull request merged**